### DISCUSSION TITLE
Question about zero downtime flow and database migrations (Laravel)

### DISCUSSION THREAD

**Discussion by binaryfire** [deployphp/deployer#3953] [Help needed] [answered]
Hi all

According to workflow in [the docs](https://deployer.org/docs/7.x/recipe/laravel), database migrations for new releases are run _before_ the new release is published and the symlinks are switched over. Which means there's a short period of time (longer if the migrations take a while...) where the old release's codebase + new release's db schema are running together. Is that correct?

If so, how do people work around the problem of old code not working with new schemas? For example a new release may drop a column that the old release is still trying to use.

Cheers

**Comment by antonmedv** [comment, marked as answer]
If your migration is non compatible with previous version on the code, you will have downtime.  

Design your migrations to be compatible and deploy ready.  Some times it makes sense to split migration into two parts and deploy twice.

**Reply by binaryfire** [reply]
@antonmedv Makes sense. Thanks!

**Comment by carlos-reynosa** [comment]
Yeah but if you're running setup upgrade the site is forced to be put into maintenance mode using the old release so I don't think it should be a problem.  I don't think you can have zero down time if your configuration has changed or if you're database needs changes.
